addons/web/static/tests/views/view_dialogs/form_view_dialog.test.js

```diff
@@ -382,3 +384,114 @@ test("Save a FormViewDialog when a required field is empty don't close the dialo
     await animationFrame();
     expect(".modal").toHaveCount(0, { message: "modal should be closed" });
 });
+
+test.tags("desktop")("new record has an expand button", async () => {
+    Partner._views["form,false"] = /* xml */ `<form><field name="foo"/></form>`;
+    const numberOfRecordsBeforeSave = Partner._records.length;
```

Comment:
I would erase Partner._records here (like you do in the last test, but with empty list here), s.t. you have full control, and you can assert that the id is 1.